tests/integration_tests/functional/test_balloon.py

```diff
@@ -0,0 +1,687 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests for guest-side operations on /balloon resources."""
+
+import logging
+import os
+import platform
+import subprocess
+import time
+
+import pytest
+
+from conftest import _test_images_s3_bucket
+from framework.artifacts import ArtifactCollection, ArtifactSet
+from framework.matrix import TestMatrix, TestContext
+from framework.builder import MicrovmBuilder, SnapshotBuilder, SnapshotType
+
+import host_tools.network as net_tools  # pylint: disable=import-error
+
+
+def get_free_mem_ssh(ssh_connection):
```

Comment:
I would move this to the utils.py file. It might make sense in other tests as well.

Reply:
Done.